core/domain/stats_domain.py

```diff
@@ -587,31 +583,35 @@ def __init__(self, calculation_output_type):
 
 class AnswerFrequencyList(AnswerCalculationOutput):
     """Domain object that represents a list of AnswerOccurrences."""
-    def __init__(self, answer_occurrences=None, limit=None):
+    def __init__(self, answer_occurrences):
         """Initialize domain object for answer frequency list for a given list
         of AnswerOccurrence objects (default is empty list).
         """
         super(AnswerFrequencyList, self).__init__(
             CALC_OUTPUT_TYPE_ANSWER_FREQUENCY_LIST)
-        answer_occurrences = answer_occurrences or []
-        self._answer_counter = utils.OrderedCounter(
-            answer_occurrence.to_pair()
-            for answer_occurrence in answer_occurrences)
-        self.limit = limit
-
-    def add_answer(self, answer):
+        self.answer_counter = utils.OrderedCounter()
+        for answer_occurrence in answer_occurrences:
+            self.answer_counter.update({
+                answer_occurrence.hashable_answer: answer_occurrence.frequency
+            })
+        self._len_limit = None
```

Comment:
I still don't think this should be part of the AnswerFrequencyList. Semantically, the list is a list of answers; it doesn't have a limit. The limit only applies to methods that query the content of the list, similar to what we might do when getting a list of entities from the datastore matching certain criteria.

Reply:
Hmm, how about I move it to `to_raw_type()`, then? That seems like a good place?

Reply:
In terms of naming, to_raw_type() suggests converting the _entire_ list to its raw form. Perhaps add another method along the lines of `get_top_answers()` or `get_top_answers_by_frequency()`?